tests/_test_utils/test_coordinate.py

```diff
@@ -39,156 +37,127 @@
+        """
+        distance = coordinate.get_distance(lat0, lon0, lat1, lon1)
+        assert int(distance) == expected_distance
+
+    @pytest.mark.parametrize("lat, lon, expected_location", [
+        (50.92, 6.36, ([50.92, 6.36], 'Juelich')),
+        (50.9200002, 6.36, ([50.92, 6.36], 'Juelich')),
+    ])
+    def test_find_location(self, lat, lon, expected_location):
+        """
+        Test finding the location from coordinates.
+        """
+        assert find_location(lat, lon) == expected_location
 
 
 class TestProjections:
+    """
+    Tests for handling coordinate projections.
+    """
+
     def test_get_projection_params(self):
+        """
+        Test fetching projection parameters for various EPSG codes.
+        """
         assert get_projection_params("epsg:4839") == {'basemap': {'epsg': '4839'}, 'bbox': 'meter(10.5,51)'}
-        with pytest.raises(ValueError):
-            get_projection_params('auto2:42005')
-        with pytest.raises(ValueError):
-            get_projection_params('auto:42001')
-        with pytest.raises(ValueError):
-            get_projection_params('crs:83')
+        for invalid_code in ['auto2:42005', 'auto:42001', 'crs:83']:
+            with pytest.raises(ValueError):
+                get_projection_params(invalid_code)
 
 
 class TestAngles:
     """
-    tests about angles
+    Tests for angle normalization and point rotation.
     """
 
-    def test_normalize_angle(self):
-        assert coordinate.fix_angle(0) == 0
-        assert coordinate.fix_angle(180) == 180
-        assert coordinate.fix_angle(270) == 270
-        assert coordinate.fix_angle(-90) == 270
-        assert coordinate.fix_angle(-180) == 180
-        assert coordinate.fix_angle(-181) == 179
-        assert coordinate.fix_angle(420) == 60
-
-    def test_rotate_point(self):
-        assert coordinate.rotate_point([0, 0], 0) == (0.0, 0.0)
-        assert coordinate.rotate_point([0, 0], 180) == (0.0, 0.0)
-        assert coordinate.rotate_point([1, 0], 0) == (1.0, 0.0)
-        assert coordinate.rotate_point([100, 90], 90) == (-90, 100)
+    @pytest.mark.parametrize("angle, normalized", [
+        (0, 0), (180, 180), (270, 270), (-90, 270), (-180, 180), (-181, 179), (420, 60),
+    ])
+    def test_normalize_angle(self, angle, normalized):
+        """
+        Test normalizing angles to the range [0, 360).
+        """
+        assert coordinate.fix_angle(angle) == normalized
+
+    @pytest.mark.parametrize("point, angle, rotated_point", [
+        ([0, 0], 0, (0.0, 0.0)),
+        ([0, 0], 180, (0.0, 0.0)),
+        ([1, 0], 0, (1.0, 0.0)),
+        ([100, 90], 90, (-90.0, 100.0)),
+        ([1.5, 0.5], 90, (-0.5, 1.5)),
+        ([0.0, 2.5], 45, (-1.7678, 1.7678)),
+    ])
+    def test_rotate_point(self, point, angle, rotated_point):
+        """
+        Test rotating points around the origin.
+        """
+        assert coordinate.rotate_point(point, angle) == pytest.approx(rotated_point)
```

Comment:
looks like you found a bug by the ([0.0, 2.5], 45, (-1.7678, 1.7678)) example

The result matches when the angle is a float